src/main/java/com/connexta/store/adaptors/S3StorageAdaptor.java

```diff
@@ -154,6 +155,21 @@ public String getStatus(@NotBlank String key) throws DatasetNotFoundException {
     return statusTag.get().getValue();
   }
 
+  public void delete(@NotBlank String key) throws QuarantineException {
+    log.info("Deleting item in bucket \"{}\" with key \"{}\"", bucket, key);
+    try {
+      if (!amazonS3.doesBucketExistV2(bucket)) {
```

Comment:
✏️ You can replace the if statements with https://github.com/connexta/ion-store/blob/e007a56b48aa6476955357372496ba9061fa6b18/src/main/java/com/connexta/store/adaptors/S3StorageAdaptor.java#L179